tests/test_gpu_openmp.py

```diff
@@ -1,15 +1,33 @@
 import numpy as np
+import pytest
 from sympy import cos
 
 from conftest import skipif
 from devito import (Grid, Dimension, Function, TimeFunction, Eq, Inc, solve,
                     Operator, switchconfig, norm)
-from devito.ir.iet import retrieve_iteration_tree
+from devito.ir.iet import Block, FindNodes, retrieve_iteration_tree
+from devito.mpi.routines import IrecvCall, IsendCall
 from examples.seismic import TimeAxis, RickerSource, Receiver
 
 
 class TestCodeGeneration(object):
 
+    @pytest.mark.parallel(mode=1)
+    @switchconfig(platform='nvidiaX')
+    def test_gpu_direct(self):
```

Comment:
this test should go at the end of the class not at the beginning, since it's a more advanced test...